### PR TITLE
Task/72 account names mosaic names endpoints

### PR DESCRIPTION
**Pull request by fboucquez**
Hi @rg911 @fullcircle23 ,

This PR will implement https://github.com/nemtech/nem2-sdk-java/issues/72

Let me know how it looks.

Cheers
Fernando

### CODE REVIEW

**Comment by fboucquez** [comment]
Hi Ravi,

I've added JavaDoc to the constructor and private repository methods.

I have also changed the AccountNames and MosaicNames model objects to use a list of NamespaceName instead of a list of just strings. This is to match the typescript implementation.

What do you think?

File: sdk-core/src/main/java/io/nem/sdk/api/AccountRepository.java [hunk 1]
```diff
@@ -49,6 +50,15 @@
...
     Observable<List<AccountInfo>> getAccountsInfo(List<Address> addresses);
 
+    /**
+     * Gets AccountNames for different accounts. The names are namespaces linked using address
+     * aliases.
+     *
+     * @param addresses List of Address
+     * @return Observable of List<{@link AccountNames}>
+     */
+    Observable<List<AccountNames>> getAccountsNames(List<Address> addresses);
```
Comment (rg911) [inline sdk-core/src/main/java/io/nem/sdk/api/AccountRepository.java]: from the rest endpoint definition, accountName takes `accountIds` as body param which is either `Addresses` or `publicKeys`. Might need to revisit it

File: sdk-core/src/main/java/io/nem/sdk/model/namespace/NamespaceId.java [hunk 2]
```diff
@@ -116,21 +116,20 @@ public String getIdAsHex() {
         return fullName;
     }
 
-    /**
-     * Compares namespaceIds for equality.
-     *
-     * @return boolean
-     */
     @Override
     public boolean equals(Object o) {
```
Comment (rg911) [inline sdk-core/src/main/java/io/nem/sdk/model/namespace/NamespaceId.java]: we should use meaning for param name instead

File: sdk-core/src/test/java/io/nem/sdk/model/mosaic/MosaicNamesTest.java [hunk 3]
```diff
@@ -18,20 +18,22 @@
...
+import java.util.List;
 import org.junit.jupiter.api.Test;
 
-class MosaicNameTest {
+class MosaicNamesTest {
 
     @Test
-    void createAMosaicName() {
-        NamespaceId namespaceId = new NamespaceId(new BigInteger("-8884663987180930485"));
+    void createAMosaicNames() {
```
Comment (rg911) [inline sdk-core/src/test/java/io/nem/sdk/model/mosaic/MosaicNamesTest.java]: typo `A` here?

File: sdk-okhttp-client/src/main/java/io/nem/sdk/infrastructure/okhttp/AccountRepositoryOkHttpImpl.java [hunk 4]
```diff
@@ -90,6 +94,28 @@ private String getAddressEncoded(String address) throws DecoderException {
...
+            .getAccountsNames(accountIds);
+        return exceptionHandling(
+            call(callback).map(AccountsNamesDTO::getAccountNames).flatMapIterable(item -> item)
+                .map(this::toAccountNames).toList().toObservable());
+    }
+    /**
+     * Converts a {@link AccountNamesDTO} into a {@link AccountNames}
+     *
+     * @param dto {@link AccountNamesDTO}
+     * @return a {@link AccountNames}
```
Comment (rg911) [inline sdk-okhttp-client/src/main/java/io/nem/sdk/infrastructure/okhttp/AccountRepositoryOkHttpImpl.java]: remove `a`?

File: sdk-vertx-client/src/main/java/io/nem/sdk/infrastructure/vertx/AccountRepositoryVertxImpl.java [hunk 5]
```diff
@@ -82,6 +86,30 @@ private String getAddressEncoded(String address) throws DecoderException {
...
+            call(callback).map(AccountsNamesDTO::getAccountNames).flatMapIterable(item -> item)
+                .map(this::toAccountNames).toList()
+                .toObservable());
+    }
+
+    /**
+     * Converts a {@link AccountNamesDTO} into a {@link AccountNames}
+     *
+     * @param dto {@link AccountNamesDTO}
+     * @return a {@link AccountNames}
```
Comment (rg911) [inline sdk-vertx-client/src/main/java/io/nem/sdk/infrastructure/vertx/AccountRepositoryVertxImpl.java]: remove `a`?

File: sdk-core/src/main/java/io/nem/sdk/api/AccountRepository.java (hunk 1 above)
Comment (fboucquez) [reply sdk-core/src/main/java/io/nem/sdk/api/AccountRepository.java]: getAccountsInfo endpoint also accepts publicKeys. Should we change that? What about having 2  methods for each endpoint? Once that accepts addresses and one that accepts public keys?

File: sdk-core/src/main/java/io/nem/sdk/model/namespace/NamespaceId.java (hunk 2 above)
Comment (fboucquez) [reply sdk-core/src/main/java/io/nem/sdk/model/namespace/NamespaceId.java]: Do you want to javadoc each equals and hashcode methods?  I use IntelliJ refactoring feature for those methods.

File: sdk-core/src/main/java/io/nem/sdk/api/AccountRepository.java (hunk 1 above)
Comment (fboucquez) [reply sdk-core/src/main/java/io/nem/sdk/api/AccountRepository.java]: I have added Added getAccountsInfoFromAddresses, getAccountsInfoFromPublicKeys, getAccountsNamesFromAddresses and getAccountsNamesFromPublicKeys. getAccountsInfoFromPublicKeys wasn't workin on the old server 5. Unsure in the new server 6. There are integration tests for all the methods.

**Pull request merged**